src/frontend/hooks/useGameInfo.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  GameInfo,
+  HyperPlayInstallInfo,
+  InstallPlatform,
+  Runner
+} from 'common/types'
+import { GogInstallInfo } from 'common/types/gog'
+import { LegendaryInstallInfo } from 'common/types/legendary'
+import { getGameInfo, getInstallInfo } from 'frontend/helpers'
+import { useEffect, useState } from 'react'
+
+type InstallInfo =
+  | GogInstallInfo
+  | LegendaryInstallInfo
+  | HyperPlayInstallInfo
+  | null
+
+interface UseGameInfoProps {
+  appName: string
+  runner: Runner
+  platform?: InstallPlatform
+  channel?: string
+}
+
+export const useGameInfo = ({
+  appName,
+  runner,
+  platform,
+  channel
+}: UseGameInfoProps) => {
+  const [gameInfo, setGameInfo] = useState<GameInfo | null>(null)
+  const [installInfo, setInstallInfo] = useState<InstallInfo | null>(null)
+
+  useEffect(() => {
+    const getNewInfo = async () => {
+      const newInfo = (await getGameInfo(appName, runner)) as GameInfo
+
+      if (newInfo && platform) {
+        const installInfo = await getInstallInfo(
+          appName,
+          newInfo.runner,
+          platform,
+          channel || 'main'
+        )
+        setInstallInfo(installInfo)
+      }
+
+      if (newInfo) {
+        setGameInfo(newInfo)
+      }
+    }
+    getNewInfo()
+  }, [appName, runner, platform, channel])
+
+  return { gameInfo, installInfo }
```

Comment:
This can be a query, you get pending states, caching, and error handling 🙏

```tsx
const query = useQuery({
    queryKey: ['gameData', appName, runner, platform, channel],
    queryFn: async () => {
      const gameInfo = await getGameInfo(appName, runner);
      
      let installInfo: InstallInfo = null;
      if (gameInfo && platform) {
        installInfo = await getInstallInfo(
          appName,
          gameInfo.runner,
          platform,
          channel
        );
      }

      return { gameInfo, installInfo };
    },
  });

  return {
    ...query.data,
    isLoading: query.isLoading,
    isError: query.isError,
    error: query.error,
  };
  ```

Reply:
I knew it!

Reply:
done the change